documentation/manual/src/doc/guides/json-codecs.md

```diff
@@ -2,25 +2,82 @@
 
 The [JSON entities algebras](/algebras/json-entities.md) show that several options
 are available to define endpoints having JSON entities. In this guide, we show
-which options should be used in which case, and which interpreters are compatible
+which option should be used in which case, and which interpreters are compatible
 with which algebras.
 
-## Use codecs at the algebra level
+The following diagram summarizes which algebra to use in which case:
 
-The most general algebra for defining JSON entities uses separate
-types for request and response entities:
+~~~ mermaid
+graph TB
+Doc(Do I want to publish documentation?)-->|yes| JsonSchemaEntities
+Doc -->|no| Codec(Do I want to implement both a client and a server?)
+Codec -->|yes| JsonEntitiesFromCodec
+Codec -->|no| JsonEntities
 
-~~~ scala src=../../../../../algebras/algebra/src/main/scala/endpoints/algebra/JsonEntities.scala#request-response-types
+style JsonSchemaEntities fill:#fff
+style JsonEntitiesFromCodec fill:#fff
+style JsonEntities fill:#fff
 ~~~
```

Comment:
This diagram looks like the following:

![questions](https://user-images.githubusercontent.com/332812/64848170-efa68800-d610-11e9-9b6a-a8ee5512342f.png)

Reply:
Ho... Nice!